interfaces/builtin/greengrass_support.go

```diff
@@ -128,11 +128,12 @@ mount options=(rw, bind) /dev/null -> /proc/sched_debug,
 mount options=(rw, bind) /dev/null -> /proc/timer_stats,
 
 # perform the pivot_root into the overlay
-pivot_root oldroot=/var/snap/greengrass/x1/rootfs/.pivot_root*/ /var/snap/greengrass/*/rootfs/,
+pivot_root oldroot=/var/snap/greengrass/@{SNAP_REVISION}/rootfs/.pivot_root*/ /var/snap/greengrass/*/rootfs/,
```

Comment:
greengrass is proprietary software and fails without this access. It relies upon various features to setup its own sandbox.

However, I'm aware of mediation issues regarding overlayfs and pivot root (in particular https://bugs.launchpad.net/apparmor/+bug/1703692), which is just one of many reasons why this interface is a 'super-privileged' interface, like docker-support and lxd-support (ie, you need a snap declaration to even install a snap that plugs this).